core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1900,7 +1900,20 @@ private[spark] object Utils extends Logging {
    */
   def resolveURI(path: String): URI = {
     try {
-      val uri = new URI(path)
+      val osSafePath = if (Path.isWindowsAbsolutePath(path, false)) {
+        // Make sure C:/ part becomes /C/.
+        val windowsUri = new URI(path)
+        val driveLetter = windowsUri.getScheme
+        s"/$driveLetter/${windowsUri.getSchemeSpecificPart()}"
+      } else if (Path.isWindowsAbsolutePath(path, true)) {
+        // Make sure /C:/ part becomes /C/.
+        val windowsUri = new URI(path.substring(1))
+        val driveLetter = windowsUri.getScheme
+        s"/$driveLetter/${windowsUri.getSchemeSpecificPart()}"
```

Comment:
So, I am a bit confused I should still change R path.

```scala
new Path(path, "rMetadata").toString
```

is being used pretty much in those wrappers. 

Would this make sense if this case (handling `C:/../..`)  is treated within `Utils.resolveURI` cc - @sarutak ? If so, I will try to neat up the codes more.